source/common/access_log/access_log_impl.cc

```diff
@@ -255,6 +258,38 @@ Grpc::Status::GrpcStatus GrpcStatusFilter::protoToGrpcStatus(
   return static_cast<Grpc::Status::GrpcStatus>(status);
 }
 
+MetadataFilter::MetadataFilter(const envoy::config::accesslog::v3::MetadataFilter& filter_config)
+    : default_res_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(filter_config, match_if_key_not_found, true)) {
+
+  auto& matcher_config = filter_config.matcher();
+
+  for (const auto& seg : matcher_config.path()) {
+    path_.push_back(seg.key());
+  }
+
+  const auto& val = matcher_config.value();
+  value_matcher_ = Matchers::ValueMatcher::create(val);
+
+  auto present_val = envoy::type::matcher::v3::ValueMatcher();
+  present_val.set_present_match(true);
+  present_matcher_ = Matchers::ValueMatcher::create(present_val);
+
+  filter_ = matcher_config.filter();
+}
+
+bool MetadataFilter::evaluate(const StreamInfo::StreamInfo& info, const Http::RequestHeaderMap&,
+                              const Http::ResponseHeaderMap&,
+                              const Http::ResponseTrailerMap&) const {
+  const auto& value =
+      Envoy::Config::Metadata::metadataValue(&info.dynamicMetadata(), filter_, path_);
+  if (present_matcher_->match(value)) {
+    return value_matcher_->match(value);
```

Comment:
Sorry to keep coming back to this, but this doesn't make much sense to me. How can a value matcher possibly match if the present matcher doesn't match? Can you add more comments or possibly fix the underlying code to somehow handle this?

Reply:
I think the logic is correct here. If the value is present we check that it matches the desired value. If it is not present we return the desired default. Or am I missing something?

Reply:
Oh I see, it's because of the default return value if not present, got it. Can you add more comments? LGTM.